beacon-chain/db/kv/slasher.go

```diff
@@ -353,11 +355,62 @@ func (s *Store) PruneAttestations(ctx context.Context, currentEpoch types.Epoch,
 	})
 }
 
+// HighestAttestations retrieves the last attestation data from the database for all indices.
+func (s *Store) HighestAttestations(
+	ctx context.Context,
+	indices []types.ValidatorIndex,
+) ([]*slashpb.HighestAttestation, error) {
+	if len(indices) == 0 {
+		return nil, nil
+	}
+
+	// Sort indices to keep DB interactions short.
+	sort.SliceStable(indices, func(i, j int) bool {
+		return uint64(indices[i]) < uint64(indices[j])
+	})
+
+	history := make([]*slashpb.HighestAttestation, 0, len(indices))
+
+	err := s.db.View(func(tx *bolt.Tx) error {
+		attBkt := tx.Bucket(attestationRecordsBucket)
+		for i := 0; i < len(indices); i++ {
+			encIdx, err := indices[i].MarshalSSZ()
```

Comment:
Can you do these outside of the view transaction? Anything to reduce the open View transaction would be ideal

Reply:
Specifically, set it up like this:
```go
encodedIndices := make([][]byte, len(indices))
for i, idx := range indices {
  encodedIndices[i], err = idx.MarshalSSZ()
  if err != nil {
   return err
   }
}
```

Then just use encodedIndices

Reply:
good point